Exercise_3.py

```diff
@@ -17,16 +19,77 @@ def append(self, data):
         Insert a new element at the end of the list.
         Takes O(n) time.
         """
-        
+        new_node=ListNode(data)
+        if self.head is None: 
+            self.head = new_node 
+        else: 
+           end = self.head 
+           while (end.next): 
+               end = end.next
+           end.next =  new_node 
+            
     def find(self, key):
         """
         Search for the first element with `data` matching
         `key`. Return the element or `None` if not found.
         Takes O(n) time.
         """
+        temp=self.head
+        if temp==None:
+            return None
+        while temp!=None:
+            if temp.data==key:
+                return temp.data
```

Comment:
Garima, What will the function return If the list exists but the key to find is not in the list?